bootstrap.sh

```diff
@@ -1,4 +1,19 @@
 #!/usr/bin/env bash
 
-# Atm this is easy... it becomes harder once we remove lark
+set -ex
+
+# Attempt to get the current location. Prefer a branch name if available,
+# otherwise fall back to the current commit's hash if we are in "detached HEAD"
+# state.
+git_location=$(git branch --show)
+if [[ "$git_location" == "" ]]; then
+    git_location=$(git rev-parse HEAD)
+fi
+
+# Stage 1; last commit where the Lark parser can parse the native parser.
+git checkout bootstrap/1
 ./glang ./parser/parser.c3 -o ./parser/parser --bootstrap -O3
+
+# Stage 2; build native parser at current commit
+git checkout "$git_location"
+./glang ./parser/parser.c3 -o ./parser/parser -O3
```

Comment:
So after this PR is merged we'll need to iterate through all the commits that break the parser as a number of unique bootstrap stages right?

Will we need to mark each breaking change with a tag? If so can we catch it on CI when we forget to mark a change as breaking

Reply:
Yeah, so I've tagged the commit *before* the breaking change (chained `or` operators in `parser.c3`) as `bootstrap/1`. That's the last commit where the Lark parser can parse the native one. Once we build the native parser, which has proper support for logical operators (you fixed the order-of-operations issue already), we can switch back to a newer version of the native parser, and build that with the stage 1 binary.

To summarise:
- Lark parser used to build old native parser (`bootstrap/1` tag)
- Old native parser (`bootstrap/1` tag) used to build latest parser 

In the future we might also need an extra step at the end:
- Latest parser used to build latest parser

> If so can we catch it on CI when we forget to mark a change as breaking

The CI will always complain if bootstrapping doesn't work, since it bootstraps the compiler every time it runs the test suite.